### ISSUE TITLE
System.InvalidProgramException occurs in a special situation that use the ternary operator.

### ISSUE THREAD

**Issue by lextudio-support** [obfuscar/obfuscar#514]
#Conditions

Problems in the following combinations.

* The ternary operation that returns a String (condition ? string1 : string2)
* The variadic function like String.Format (string, params string[])
* The assignment to a property (string Property { get; set; })

#Environment

* Windows 7 Professional(64bit)
* Microsoft Visual C# 2010 Express
* Obfuscar 2.0 rc7

#Example Source
##Program.cs
```
namespace Example
{
    public class Model
    {
        public string Property { get; set; }
    }

    public class Program
    {
        public static void Main(string[] args)
        {
            InvalidFunction(true);
        }

        public static void InvalidFunction(bool flag)
        {
            Model model = new Model();
            model.Property = flag ? MultiParameterFunction("example") : "test";
        }

        public static string MultiParameterFunction(string value, params string[] args)
        {
            return value;
        }
    }
}
```
##Obfuscar.xml
```
<?xml version='1.0'?>

<Obfuscator>
  <Var name="InPath" value=".\" />
  <Var name="OutPath" value=".\Out" />

  <Module file="$(InPath)\Example.exe" />
</Obfuscator>
```

#Execution Result
```
>Example.exe

Unhandled Exception: System.InvalidProgramException: Common Language Runtime detected an invalid program.
   at A.a.A(Boolean )
   at A.a.A(String[] )
```

#Workaround

I stopped using the ternary operator.
I was using the if statement instead.

**Original Reported Date**: 2015-02-25T18:15:34.873-08:00
**Planned For Release**: 
**Original CodePlex ID**: 26

**Comment by lextudio-support** [comment]
**Copied from CodePlex without authors**:

Thanks for reporting this. I can see it is a bug affecting 2.0 and 2.0.1.

You can try out 2.0.2 Beta 1 as this issue seems to be fixed in recent changes.

**Original Posted Date**: 2015-04-17T23:18:38.97-07:00

**Comment by lextudio-support** [comment]
**Original Closed Date**: 2015-04-17T23:18:53.643-07:00
